catlearn/ga/algorithm.py

```diff
@@ -40,26 +40,34 @@ def __init__(self, population_size, fit_func, features, targets,
             mutation operations.
         fitness_parameters : int
             The number of variables to optimize. Default is a single variable.
-        nslpit : int
+        nsplit : int
             Number of data splits for k-fold cv.
         accuracy : int
             Number of decimal places to include when finding unique candidates
             for duplication removal. If None, duplication removel is not
             performed.
         """
         # Set parameters.
+        self.nprocs = nprocs
         self.step = -1
-        self.population_size = population_size
         self.fit_func = fit_func
         self.dimension = features.shape[1]
         self.nsplit = nsplit
         self.accuracy = accuracy
 
+        if population_size == 'auto':
+            population_size = 2 * self.nprocs * ((16 // self.nprocs) + 1)
+        elif isinstance(population_size, int):
```

Comment:
Shouldn't this just be `if` instead of `elif`? The `if population_size == 'auto'` line makes `population_size` an `int` which can always be assigned to `self.population_size`. Isn't this missed at the moment? Or does it still pass with `auto`?